src/preSaveHandlers/handlers/dateColumnHandler.js

```diff
@@ -13,17 +30,79 @@ class DateColumnHandler extends PreSaveHandler {
      * @param {PreSaveAsyncProps}
      * @returns {Promise<import('../../fhir/classes/4_0_0/resources/resource')>}
      */
+    // if a column is of date type then set it to date (if not already)
     async preSaveAsync ({ resource }) {
-        for (const [fieldName, field] of Object.entries(resource)) {
-            // if a column is of date type then set it to date (if not already)
-            // TODO: this currently only handles one level deep fields.  Change it to handle fields multiple levels deep
-            if (isColumnDateType(resource.resourceType, fieldName) && field) {
-                if (!(resource[`${fieldName}`] instanceof Date)) {
-                    resource[`${fieldName}`] = new Date(field);
+        await this.processResource(resource, '');
+        return resource;
+    }
+
+    async processResource (resource, path) {
+        const extendPath = (existingPath, key) => {
+            return existingPath ? `${existingPath}.${key}` : key;
+        };
+
+        const recurseData = (object, parentPath) => {
+            Object.keys(object).forEach(key => {
+                const currentPath = extendPath(parentPath, key);
+                const value = object[key];
+                if (Array.isArray(value)) {
+                    value.forEach((item, index) => {
+                    if (typeof item === 'object' && item !== null) {
+                        recurseData(item, extendPath(currentPath, index));
+                    } else {
+                        if (this.shouldUpdate(resource, extendPath(currentPath, index))) {
+                            object[key][index] = this.setDate(item);
+                        }
+                    }
+                });
```

Comment:
Can we replace this with recurseData(value, currentPath)? I think that might work the same, also simplifying the code

Reply:
Tried switching that out, and dates in Arrays are not addressed, so I think we should leave as is.